extension/src/popup/components/sendPayment/SendConfirm/TransactionDetails/index.tsx

```diff
@@ -269,44 +267,19 @@ export const TransactionDetails = ({ goBack }: { goBack: () => void }) => {
 
   const handleXferTransaction = async () => {
     try {
-      const assetAddress = asset.split(":")[1];
-      const assetBalance = tokenBalances.find(
-        (b) => b.contractId === assetAddress,
-      );
-
-      if (!assetBalance) {
-        throw new Error("Asset Balance not available");
-      }
-
-      const parsedAmount = parseTokenAmount(
-        amount,
-        Number(assetBalance.decimals),
-      );
-
-      const params = [
-        new SorobanClient.Address(publicKey).toScVal(), // from
-        new SorobanClient.Address(destination).toScVal(), // to
-        new SorobanClient.XdrLargeInt("i128", parsedAmount.toNumber()).toI128(), // amount
-      ];
-
       if (!hasSorobanClient(sorobanClient)) {
         throw new Error("Soroban RPC not supported for this network");
       }
```

Comment:
do we still need this check? We're not actually using `sorobanClient` (lowercase) anymore since we've already built and simmed the tx earlier

Reply:
we do end up using it in the following action [for refreshing balances](https://github.com/stellar/freighter/blob/61277e9d7d582ca2cc41daa6bc828ca4557207e3/extension/src/popup/ducks/transactionSubmission.ts#L147C13-L147C26). It's kind of annoying to have to pass in the client just for the refresh so maybe it would be nice to refactor those actions to build their own clients or something?

Reply:
Oh gotcha. Yeah, maybe we refactor those to use their own clients later on 👍